[PR] MSI: 512: MFTF Tests for MSI - Part 1

ivy00johns opened the pull request:
### Description
This is the first set of Test Cases created for MSI using MFTF. This PR contains 8 MSI specific Tests that take ~4 mins to run locally. Please reference the following Wiki for instructions on how to run them: https://github.com/magento-engcom/msi/wiki/Getting-Started

### PLEASE NOTE
- These Tests will **NOT** generate due to Page Objects that I added to the private Magento2CE repo. I'm working on getting them merged. Once that is merged the Tests will generate. PR: https://github.com/magento-pangolin/magento2ce/pull/37
- These Test will **NOT** run currently due to a Bug we discovered in MFTF, MQE-820, once the bug fix is merged these Tests will run.

### Related Pull Request
https://github.com/magento/magento2ce/pull/2280

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

ivy00johns committed:
782: MST MFTF Test Cases

- Initial commit of MFTF Tests.

ivy00johns committed:
782: MST MFTF Test Cases

- Adding additional data to the _defaultSource model.
- Replacing "TestSource1" with "_defaultSource".
- Removing test group.
- Adding "clearGridFilterIfPresent" to clean up before the Test runs.
- Adding SearchByKeyword test case.

ivy00johns committed:
782: MST MFTF Test Cases

- Adding Stock Metadata.
- Adding Stock Page.
- Adding Stock Test.
- Renamed a few Sections.
- Updated Tests to use new names.

ivy00johns committed:
782: MST MFTF Test Cases

- Adding the "Assign Sources" slide out section.

ivy00johns committed:
782: MST MFTF Test Cases

- Adding Stock meta-data.
- Renaming AdminManageStockGrid. Adding additional selectors.
- Renamed a few Test files.
- Adding Search by Keyword test for Stock.

ivy00johns committed:
782: MST MFTF Test Cases

- Adding additional Stock data.
- Adding UpdateStockDataTest.

ivy00johns committed:
782: MST MFTF Test Cases

- Removing references to Priority since it was removed from the UI.
- Replacing "_defaultSource" reference with "TestSource1".
- Updating the dataType for the Metadata/Data.

ivy00johns committed:
782: MST MFTF Test Cases

- Fixed the Title of a Test.
- Adding scrollToTopOfPage to a Test that had issues with the screen size.

ivy00johns committed:
782: MST MFTF Test Cases

- Replacing actions with the new Action Groups I setup.

ivy00johns committed:
782: MSI MFTF Test Cases

- Adding new attribute to the metadata so the URL is generated correctly for the Form Post.
- Fixed an invalid element "type" for a few selectors.
- Replacing old action group with a conditional click for the "Clear All" button.
- Moved all Tests to the "acceptance" directory instead of "functional".

ivy00johns committed:
782: MSI MFTF Test Cases

- Updating metadata URL attribute name to match newest schema changes.

ivy00johns committed:
782: MSI MFTF Test Cases

- Fixed a selector. It was missing a "."
- Updated the Stock Data name.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Metadata/source-meta.xml:
```diff
@@ -0,0 +1,37 @@
+
+<operations xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="../../../../../../vendor/magento/magento2-functional-testing-framework/src/Magento/FunctionalTestingFramework/DataGenerator/etc/dataOperation.xsd">
+    <operation name="CreateSource" dataType="source" type="create"
```
Please create operation which will create `Source` entity via WebAPI.
Also please move this operation to `InventoryApi` module

ivy00johns replied:
I moved the "Data" and "Metadata" folders to the InventoryApi module.

ivy00johns replied:
Converted the AdminForm post to an API post.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Metadata/stock-meta.xml:
```diff
@@ -0,0 +1,18 @@
+
+<operations xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="../../../../../../vendor/magento/magento2-functional-testing-framework/src/Magento/FunctionalTestingFramework/DataGenerator/etc/dataOperation.xsd">
+    <operation name="CreateStock" dataType="stock" type="create"
```
Please create operation which will create `Stock` entity via WebAPI.
Please move this operation to `InventoryApi` module

ivy00johns replied:
Converted the AdminForm post to an API post.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Data/MsiSourceData.xml:
```diff
@@ -0,0 +1,84 @@
+        <data key="id_field_name">source_code</data>
+        <data key="name" unique="suffix">Test Source 2 </data>
+        <data key="source_code" unique="suffix">test_source_2_</data>
+        <data key="isEnabled">true</data>
```
Based on `\Magento\InventoryApi\Api\Data\SourceInterface` there is no field `isEnabled`, there is just `enabled`

ivy00johns replied:
I did this for a few reasons:
- I needed BOTH values for my Test, one for the metadata which requires a binary value and one for the Test Action that requires a boolean value. 
- I also wanted to show that you CAN include data keys that do NOT match the data keys for your API or Form Post metadata.

ivy00johns replied:
I removed the duplicate values.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Data/MsiSourceData.xml:
```diff
@@ -0,0 +1,84 @@
+        <data key="id_field_name">source_code</data>
+        <data key="name" unique="suffix">Test Source 1 </data>
+        <data key="source_code" unique="suffix">test_source_1_</data>
+        <data key="isEnabled">true</data>
```
Based on `\Magento\InventoryApi\Api\Data\SourceInterface` there is no field `isEnabled`, there is just `enabled`

ivy00johns replied:
I did this for a few reasons:
- I needed BOTH values for my Test, one for the metadata which requires a binary value and one for the Test Action that requires a boolean value. 
- I also wanted to show that you CAN include data keys that do NOT match the data keys for your API or Form Post metadata.
    - Added a Note to clarify.

ivy00johns replied:
I removed the duplicate values.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Data/MsiSourceData.xml:
```diff
@@ -0,0 +1,84 @@
+        <data key="name">Default Source</data>
+        <data key="source_code">default</data>
+        <data key="enabled">1</data>
+        <data key="isEnabled">true</data>
```
Based on `\Magento\InventoryApi\Api\Data\SourceInterface` there is no field `isEnabled`, there is just `enabled`

ivy00johns replied:
I did this for a few reasons:
- I needed BOTH values for my Test, one for the metadata which requires a binary value and one for the Test Action that requires a boolean value. 
- I also wanted to show that you CAN include data keys that do NOT match the data keys for your API or Form Post metadata.
    - Added a Note to clarify.

okolesnyk replied:
Try to leave only this line: `<data key="enabled">true</data>`

ivy00johns replied:
I removed the duplicate values.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Data/MsiSourceData.xml:
```diff
@@ -0,0 +1,84 @@
+        <data key="postcode">90230</data>
+        <data key="use_default_carrier_config">1</data>
+    </entity>
+    <entity name="TestSource1" type="source">
```
Please provide meaningful name for this entity

ivy00johns replied:
Renamed "FullSource1" instead. If that doesn't work, please recommend a name that you find applicable.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Data/MsiSourceData.xml:
```diff
@@ -0,0 +1,84 @@
+        <data key="useGlobalShippingConfiguration">true</data>
+        <data key="use_default_carrier_config">1</data>
+    </entity>
+    <entity name="TestSource2" type="source">
```
Please provide meaningful name for this entity

ivy00johns replied:
Renamed "FullSource2" instead. If that doesn't work, please recommend a name that you find applicable.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Data/MsiSourceData.xml:
```diff
@@ -0,0 +1,84 @@
+        <data key="country_id">US</data>
+        <data key="postcode">00000</data>
+        <data key="use_default_carrier_config">1</data>
+        <data key="useGlobalShippingConfiguration">true</data>
```
Please use `use_default_carrier_config` and do not duplicate data

ivy00johns replied:
I did this for a few reasons:
- I needed BOTH values for my Test, one for the metadata which requires a binary value and one for the Test Action that requires a boolean value. 
- I also wanted to show that you CAN include data keys that do NOT match the data keys for your API or Form Post metadata.
    - Added a Note to clarify.

ivy00johns replied:
I removed the duplicate values.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Data/MsiSourceData.xml:
```diff
@@ -0,0 +1,84 @@
+        <data key="city">Culver City</data>
+        <data key="street">6161 West Centinela Avenue</data>
+        <data key="postcode">90230</data>
+        <data key="useGlobalShippingConfiguration">true</data>
```
Please use `use_default_carrier_config` and do not duplicate data

ivy00johns replied:
I did this for a few reasons:
- I needed BOTH values for my Test, one for the metadata which requires a binary value and one for the Test Action that requires a boolean value. 
- I also wanted to show that you CAN include data keys that do NOT match the data keys for your API or Form Post metadata.
    - Added a Note to clarify.

ivy00johns replied:
I removed the duplicate values.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Data/MsiSourceData.xml:
```diff
@@ -0,0 +1,84 @@
+        <data key="city">Québec City</data>
+        <data key="street">1234 Test Avenue</data>
+        <data key="postcode">QC G2E 6J5</data>
+        <data key="useGlobalShippingConfiguration">true</data>
```
Please use `use_default_carrier_config` and do not duplicate data

ivy00johns replied:
I did this for a few reasons:
- I needed BOTH values for my Test, one for the metadata which requires a binary value and one for the Test Action that requires a boolean value. 
- I also wanted to show that you CAN include data keys that do NOT match the data keys for your API or Form Post metadata.

ivy00johns replied:
I removed the duplicate values.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Test/AdminCreateNewSourceTest.xml:
```diff
@@ -0,0 +1,66 @@
+        <checkOption selector="{{AdminEditSourceCarriersSection.useGlobalShippingConfiguration}}" stepKey="checkUseGlobalShippingConfiguration1"/>
+        <click selector="{{AdminProductFormActionSection.saveArrow}}" stepKey="clickOnSaveArrow1"/>
+        <click selector="{{AdminProductFormActionSection.saveAndClose}}" stepKey="clickOnSaveAndClose"/>
+
```
Please add all assertions listed in scenario.

1. Confirmation message "The Source has been saved" appears
2. Verify that data in all tabs is correct

ivy00johns replied:
Test updated.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Test/AdminCreateNewStockWithNoSourceOrSalesChannelTest.xml:
```diff
@@ -0,0 +1,41 @@
+
+        <fillField userInput="{{_defaultStock.name}}" selector="{{AdminEditStockGeneralSection.name}}" stepKey="fillName"/>
+
+        <actionGroup ref="AdminSaveAndClose" stepKey="saveAndClose1"/>
```
Please add assertion that success message is appeared
https://hiptest.net/app/projects/69435/test-plan/folders/409123/scenarios/1408724

ivy00johns replied:
The Action Group "AdminSaveAndClose" contains an assert already.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Test/AdminCreateSimpleProductWithDefaultSourceTest.xml:
```diff
@@ -0,0 +1,79 @@
+        <scrollToTopOfPage stepKey="scrollToTopOfPage"/>
+        <click selector="{{AdminProductSourcesSection.assignSources}}" stepKey="clickOnAssignSource"/>
+
+        <conditionalClick selector="{{AdminGridFilterControls.clearAll}}" dependentSelector=".admin__data-grid-header[data-bind='afterRender: \$data.setToolbarNode'] .admin__data-grid-filters-current._show" visible="true" stepKey="clearTheFiltersIfPresent"/>
```
I think it is enough to have dependentSelector like this `.admin__data-grid-header .admin__data-grid-filters-current._show`

ivy00johns replied:
If you use that selector it will locate 2 elements on the page that match it, one for the floating grid filters and one for the sticky grid filters. The selector ".admin__data-grid-header[data-bind='afterRender: \$data.setToolbarNode']" is the only unique thing that I've found currently that differentiates them.

ivy00johns replied:
Selector updated.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Test/AdminCreateSimpleProductWithDefaultSourceTest.xml:
```diff
@@ -0,0 +1,79 @@
+        <click selector="{{AdminAssignSourcesSlideOutSection.done}}" stepKey="clickOnDone"/>
+
+        <fillField selector="{{AdminProductSourcesGrid.rowQty}}" userInput="{{SimpleMsiProduct.quantity}}" stepKey="fillQtyField"/>
+        <fillField selector="{{AdminProductSourcesGrid.rowNotifyQuantity}}" userInput="{{SimpleMsiProduct.notifyQuantity}}" stepKey="fillNotifyQtyField"/>
```
In test there is no step for filling `Notify Quantity`.
Please make automated test synced with documented in HipTest.

ivy00johns replied:
Removed this action, the field now defaults to "0" upon save.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Test/AdminCreateSimpleProductWithDefaultSourceTest.xml:
```diff
@@ -0,0 +1,79 @@
+        <fillField selector="{{AdminProductSourcesGrid.rowNotifyQuantity}}" userInput="{{SimpleMsiProduct.notifyQuantity}}" stepKey="fillNotifyQtyField"/>
+
+        <actionGroup ref="AdminSaveAndClose" stepKey="saveAndClose1"/>
+
```
Please add assert success message.

ivy00johns replied:
The "AdminSaveAndClose" method contains an assert already.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Test/AdminCreateSimpleProductWithDefaultSourceTest.xml:
```diff
@@ -0,0 +1,79 @@
+            <argument name="selector" value="AdminProductGridFilterControls.sku"/>
+            <argument name="value" value="SimpleMsiProduct.sku"/>
+        </actionGroup>
+
```
Please add asserts which verifies that product appears in "Products" grid and fields "Name", "SKU", "Price", "Quantity", "Status" and "Website" are correct.

ivy00johns replied:
Added logic to accomplish this in the Test.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Test/AdminFilterSourcesInGridTest.xml:
```diff
@@ -0,0 +1,43 @@
+
+        <see userInput="$$createSource3.general[source_code]$$" selector="{{AdminGridRow.rowOne}}" stepKey="seeCodeInRow1"/>
+        <see userInput="$$createSource3.general[name]$$" selector="{{AdminGridRow.rowOne}}" stepKey="seeNameInRow1"/>
+        <see userInput="Enabled" selector="{{AdminGridRow.rowOne}}" stepKey="seeIsEnabledInRow1"/>
```
Based on what is described in scenario there are should be one more step to verify filtering is working as expected)
```
13. Go to Stores -> Manage Sources
14. Click button "Filters"
15. Put in "Name" field value "Test Source 3"
16. click button "Apply filters"
17. verify that filter returned "Test Source 3" in grid
18. verify message "Active filters:	 Name: Test Source 3
19. click button "Clear all"
20. Click button "Filters"
21. Put in field "Code" value "test_sorce_4" 
22. click button "Apply filters"
23. verify that filter returned "Test Source 4" in grid
24. verify message "Active filters:	 Code: test_sorce_4 
25. Then Data filters worked correctly in Manage Sources grid
```

ivy00johns replied:
Test updated.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Test/AdminManageSourcesSearchByKeywordTest.xml:
```diff
@@ -0,0 +1,40 @@
+        <waitForPageLoad stepKey="waitForPageLoad1"/>
+
+        <actionGroup ref="AdminGridSearchByKeyword" stepKey="searchByKeyword">
+            <argument name="value" value="$$createSource2.general[name]$$"/>
```
In scenario described put only piece of source name and exactly number "4" to search by keyword field and verify that "Test Source 4" appears.

```
14. Put in "Search by keyword" field value "4"
15. verify that search returned "Test Source 4" in grid
16. Then Search by keyword worked properly in Manage Sources grid
```

ivy00johns replied:
We cannot perform these exact validation steps because:
* We cannot isolate the unique suffix that is added to the Source name when it persisted to the DB. 
    - Therefore I have to do a full word search to match the exact name which will result in a single entity in the DB.
    - IF we could purge the DB before a Test began we could do something hacky to perform this assert. 
        * Clear the DB
        * Then persist data in a know setup
        * Search for a known unique partial piece of data.
        * Assert against a brand new, known data set

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Test/AdminManageStocksSearchByKeywordTest.xml:
```diff
@@ -0,0 +1,39 @@
+        <waitForPageLoad stepKey="waitForPageLoad1"/>
+
+        <actionGroup ref="AdminGridSearchByKeyword" stepKey="searchByKeyword">
+            <argument name="value" value="$$createStock3.general[name]$$"/>
```
In scenario described put only piece of Stock name and exactly number "4" to search by keyword field and verify that "Test Stock 4" appears.
```
9. Put in "Search by keyword" field value "4"
10. verify that search returned "Test Stock 4" in grid
11. Then Search by keyword worked properly in Manage Stocks grid
```

ivy00johns replied:
We cannot perform these exact validation steps because:
* We cannot isolate the unique suffix that is added to the Stock name when it persisted to the DB. 
    - Therefore I have to do a full word search to match the exact name which will result in a single entity in the DB.
    - IF we could purge the DB before a Test began we could do something hacky to perform this assert. 
        * Clear the DB
        * Then persist data in a know setup
        * Search for a known unique partial piece of data.
        * Assert against a brand new, known data set

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Test/AdminManageStocksSearchByKeywordTest.xml:
```diff
@@ -0,0 +1,39 @@
+        <before>
+            <createData entity="_defaultStock" stepKey="createStock1"/>
+            <createData entity="_defaultStock" stepKey="createStock2"/>
+            <createData entity="_defaultStock" stepKey="createStock3"/>
```
`Given` block contains 4 Stocks created. Please make automated test synced with documented scenario in HipTest.

ivy00johns replied:
Added an additional Stock.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Test/AdminUpdateSourceDataTest.xml:
```diff
@@ -0,0 +1,90 @@
+        <conditionalClick selector="{{AdminEditSourceCarriersSection.closed}}" dependentSelector="{{AdminEditSourceCarriersSection.opened}}" visible="false" stepKey="clickOnCarriers1"/>
+        <checkOption selector="{{AdminEditSourceCarriersSection.useGlobalShippingConfiguration}}" stepKey="checkUseGlobalShippingConfiguration1"/>
+
+        <click selector="{{AdminProductFormActionSection.saveButton}}" stepKey="clickOnSaveContinue1"/>
```
I suggest to create Source entity via WebAPI then go to "Stores" -> "Manage Sources", find created Source and edit required fields then save it and perform all assertions. This will save a lot of time during test run and will not affect Test Case Scenario.

ivy00johns replied:
* Optimizing it to use the FormPost instead of the UI. This was one of the 1st tests I created before I setup the Metadata/Data.

ivy00johns replied:
Test updated.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Test/AdminUpdateSourceDataTest.xml:
```diff
@@ -0,0 +1,90 @@
+        <fillField userInput="{{TestSource2.street}}" selector="{{AdminEditSourceAddressDataSection.street}}" stepKey="fillStreetField2"/>
+        <fillField userInput="{{TestSource2.postcode}}" selector="{{AdminEditSourceAddressDataSection.postcode}}" stepKey="fillPostcodeField2"/>
+
+        <actionGroup ref="AdminSaveAndClose" stepKey="saveAndClose1"/>
```
In test scenario I see we should click "Save and Continue" and verify data in all tabs is correct.

```
18. click button "Save & Continue"
19. verify that data in all tabs is correct
20. Then verify in Manage Sources grid that Source's data changes applied properly
```

ivy00johns replied:
Test updated.

okolesnyk commented on app/code/Magento/Inventory/Test/Acceptance/Test/AdminUpdateStockDataTest.xml:
```diff
@@ -0,0 +1,44 @@
+        <click selector="{{AdminGridMainControls.add}}" stepKey="clickOnAdd"/>
+
+        <fillField userInput="{{TestStock1.name}}" selector="{{AdminEditStockGeneralSection.name}}" stepKey="fillName1"/>
+        <click selector="{{AdminGridMainControls.saveAndContinue}}" stepKey="openSaveDropDown"/>
```
Please create "Stock" entity via WebAPI and then find it in Manage Stock grid, perform editing and all needed assertions.

ivy00johns replied:
Converted the AdminForm post to an API post.

okolesnyk reviewed (changes requested):
Please take a look at the comments I wrote and please move all Functional tests to `dev/tests/acceptance` directory.

okolesnyk commented:
Static build is failing.
```
Magento\Test\Integrity\Xml\SchemaTest::testXmlFiles
Passed: 1927, Failed: 21, Incomplete: 0, Skipped: 0.
Data set: /app/code/Magento/Inventory/Test/Acceptance/Page/AdminManageSourcesPage.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Page/AdminManageSourcesPage.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
Data set: /app/code/Magento/Inventory/Test/Acceptance/Page/AdminManageStockPage.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Page/AdminManageStockPage.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
Data set: /app/code/Magento/Inventory/Test/Acceptance/Metadata/stock-meta.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Metadata/stock-meta.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
Data set: /app/code/Magento/Inventory/Test/Acceptance/Metadata/source-meta.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Metadata/source-meta.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
Data set: /app/code/Magento/Inventory/Test/Acceptance/Data/MsiSourceData.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Data/MsiSourceData.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
Data set: /app/code/Magento/Inventory/Test/Acceptance/Data/MsiProductData.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Data/MsiProductData.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
Data set: /app/code/Magento/Inventory/Test/Acceptance/Data/MsiStockData.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Data/MsiStockData.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
Data set: /app/code/Magento/Inventory/Test/Acceptance/Test/AdminFilterSourcesInGridTest.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Test/AdminFilterSourcesInGridTest.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
Data set: /app/code/Magento/Inventory/Test/Acceptance/Test/AdminManageSourcesSearchByKeywordTest.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Test/AdminManageSourcesSearchByKeywordTest.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
Data set: /app/code/Magento/Inventory/Test/Acceptance/Test/AdminCreateNewStockWithNoSourceOrSalesChannelTest.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Test/AdminCreateNewStockWithNoSourceOrSalesChannelTest.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
Data set: /app/code/Magento/Inventory/Test/Acceptance/Test/AdminCreateNewSourceTest.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Test/AdminCreateNewSourceTest.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
Data set: /app/code/Magento/Inventory/Test/Acceptance/Test/AdminManageStocksSearchByKeywordTest.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Test/AdminManageStocksSearchByKeywordTest.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
Data set: /app/code/Magento/Inventory/Test/Acceptance/Test/AdminUpdateSourceDataTest.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Test/AdminUpdateSourceDataTest.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
Data set: /app/code/Magento/Inventory/Test/Acceptance/Test/AdminCreateSimpleProductWithDefaultSourceTest.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Test/AdminCreateSimpleProductWithDefaultSourceTest.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
Data set: /app/code/Magento/Inventory/Test/Acceptance/Test/AdminUpdateStockDataTest.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Test/AdminUpdateStockDataTest.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
Data set: /app/code/Magento/Inventory/Test/Acceptance/Section/AdminManageStockFormSection.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Section/AdminManageStockFormSection.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
Data set: /app/code/Magento/Inventory/Test/Acceptance/Section/AdminManageStockGridSection.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Section/AdminManageStockGridSection.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
Data set: /app/code/Magento/Inventory/Test/Acceptance/Section/AdminManageSourcesEditFormSection.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Section/AdminManageSourcesEditFormSection.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
Data set: /app/code/Magento/Inventory/Test/Acceptance/Section/AdminManageSourcesGridSection.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Section/AdminManageSourcesGridSection.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
Data set: /app/code/Magento/Inventory/Test/Acceptance/Section/AdminProductSourcesFormSection.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Section/AdminProductSourcesFormSection.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
Data set: /app/code/Magento/Inventory/Test/Acceptance/Section/AdminAssignSourcesSlideOutSection.xml
The XML file at /home/travis/build/magento-engcom/msi/app/code/Magento/Inventory/Test/Acceptance/Section/AdminAssignSourcesSlideOutSection.xml does not have a schema properly defined.  It should have a xsi:noNamespaceSchemaLocation attribute defined with a URN path.  E.g. xsi:noNamespaceSchemaLocation="urn:magento:framework:Relative_Path/something.xsd"
Failed asserting that 0 matches expected 2.
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:35
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:115
/home/travis/build/magento-engcom/msi/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:70
/home/travis/build/magento-engcom/msi/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php:48
```

ivy00johns committed:
782: MSI MFTF Test Cases 2

- Addressed CR concerns.
- Moved the API files to the correct module.
- Added addition asserts that I missed.
- Updated a URL to match the changes we made in MFTF.

ivy00johns reviewed (commented):
Issues addressed in the latest Commit to the branch. Please review.

ivy00johns committed:
782: MSI MFTF Test Cases 2

- Renamed "AdminSaveAndClose" action group to "AdminFormSaveAndClose".

ivy00johns committed:
782: MSI MFTF Test Cases 2

- Removing data that is no longer needed.

ivy00johns commented:
Working on committing fixes for Alex's comments.

ivy00johns committed:
782: MSI MFTF Test Cases 1

- Removing an incomplete test.
- Replacing the Admin Form Post metadata with an API metadata alternative.
- Updating the Tests to use the API post instead of the Form Post.

ivy00johns committed:
727: MSI Test Cases 1

- Updating selector for the Filter drop down menu and moved it to a section.
- Updated Tests to use the section element instead of being hardcoded.

ivy00johns committed:
727: MSI Test Cases 1

- Moving all Inventory Test files to /dev/tests/acceptance from /app/code/.
- Adding composer.json, license files and README to the new folders.

okolesnyk commented on dev/tests/acceptance/tests/functional/Magento/FunctionalTest/Inventory/README.md:
```diff
@@ -0,0 +1,3 @@
+# Magento 2 Acceptance Tests
```
Based on our standard we call our tests/packages Functional.
Example
https://github.com/magento/magento2ce/tree/2.3-develop/dev/tests/acceptance/tests/functional/Magento/FunctionalTest/Authorization

ivy00johns replied:
OOOPS! I copied the "Integration" module which used the words "Acceptance". I'll update my new modules.

ivy00johns replied:
Changes committed!

okolesnyk reviewed (approved):
❗️❗️❗️Please rename everything from Acceptance Tests to Functional Tests. ❗️❗️❗️
Based on our standard we call our tests/packages Functional.
Example of already created modules:
https://github.com/magento/magento2ce/tree/2.3-develop/dev/tests/acceptance/tests/functional/Magento/FunctionalTest/Authorization

**Everything else looks good!**

okolesnyk commented:
@ivy00johns FYI
As far as I know everybody in MSI project trying to stabilize/fix Functional test builds. So I assume you should link here Jenkins builds with your tests included to verify you didn't brake anything.
Based on runs on my local machine you shouldn't have any problems with this.

ivy00johns committed:
782: MSI MFTF Test Cases 1

- Replacing the word "Acceptance" with "Functional".

ivy00johns committed:
782: MSI MFTF Test Cases 1

- Updating a few Source selectors to make them parameterized.
- Replacing the "LogoutOfAdmin" Action Group with "logout".
- Replacing the "AdminGridSearchByKeyword" Action Group with "searchAdminDataGridByKeyword".
- Adding severity to all Tests.

ivy00johns committed:
782: MSI MFTF Test Cases 1

- Moving my Test and supporting files to "app/code/".

ivy00johns committed:
Merge branch '2.3-develop' into 782-MSI-MFTF-Test-Cases

vrann commented on app/code/Magento/Inventory/Test/Acceptance/LICENSE.txt:
```diff
@@ -0,0 +1,48 @@
+
+Open Software License ("OSL") v. 3.0
```
Is it really needed to have README and LICENSE for every test directory? Usually we have jsut one of these files in the root of the module

okolesnyk replied:
Yes it is. All MFTF tests will be excluded when packages will be created. Separate package will be created only with MFTF tests. And this files are going to represent License and all additional info in that separate package.

maghamed commented:
@ivy00johns Static tests build is red

ivy00johns commented:
The static build is red because MFTF does not support URN's at this time. We need to do some work on MFTF core to support them. In the meantime we need to blacklist these XML files from that build so we can merge the work and make the necessary changes in future to support URN's. Working with Alex K to get this done.

okolesnyk commented:
@ivy00johns 
Please add MFTF xml files to this filter:
https://github.com/magento/magento2ce/blob/2.3-develop/dev/tests/static/testsuite/Magento/Test/Integrity/Xml/SchemaTest.php#L103
